### ISSUE TITLE
Suggest using format precision for floats instead of casting to integer

### ISSUE THREAD

**Issue by hellow554** [rust-lang/rust-clippy#7515]
### What it does

*What does this lint do?*
Linting code where a floating point value (f64/f32) is casted to an integer just for printing/formatting. The intention behind that is to remove the fraction part of that number and only display the integer itself. This is closely related to [`cast_possible_truncation`](https://rust-lang.github.io/rust-clippy/master/index.html#cast_possible_truncation).

This kind of lint could be applied to everything, that takes a fmt string, e.g. `format`, `panic`, `write`, ... .

### Categories (optional)

`clippy::pedantic` maybe `clippy::complexity` or `clippy::suspicious`. I'm not sure here (:

### What is the advantage of the recommended code over the original code

If the value should be `NaN` or `INFINITY`, it will be printed/formatted as such and easier to detect instead of silently converting it to 0 or min/max of the type.

### Drawbacks

Could be a false positive.

### Example

```rust
fn print_it(a: f64) {
    println!("{}", a as i64);
}
```

Could be written as:

```rust
fn print_it(a: f64) {
    println!("{:.0}", a);
}
```

**Comment by hellow554** [comment]
This should of course (if possible) also apply to `i64::from(a)` conversions (and other integers)
